### PR TITLE
Hotfix: Fixed reference block number

### PR DESCRIPTION
**Pull request by alan-aboudib-mc**
**Context**

The current implementation of the dataset migration evaluates only models submitted after a specified block on the new dataset. This behavior does not align with our intended functionality.

**Goal**

The objective of this PR is to ensure that after the specified migration block, all models, regardless of their submission time, are evaluated on the new dataset. Additionally, once the migration block is reached, the code supporting the old dataset could be completely removed, streamlining the evaluation process.

### CODE REVIEW

File: neurons/validator.py [hunk 1]
```diff
@@ -588,8 +586,8 @@ async def run_step(self):
                 hotkey
             )
 
-            old_losses = [math.inf for _ in range(len(old_batches))]
```
Comment (RusticLuftig) [inline neurons/validator.py:591]: The batches will be different lengths since you load the same number of pages but use different sequence lengths.

As a higher level comment, I'd recommend first prohibiting models prior to the 7B block (reject them in the update loop) and then you can remove a lot of the complexity in this function since you won't have to handle the new and old tokenizers. If that seems too expensive to do now, then perhaps something to think about for the next release

File: neurons/validator.py [hunk 2]
```diff
@@ -658,7 +653,7 @@ async def run_step(self):
 
         # Compute wins and win rates per uid.
         wins, win_rate = pt.validation.compute_wins(
-            uids, losses_per_uid, batches_to_use, uid_to_block
+            uids, losses_per_uid, batches_new, uid_to_block
```
Comment (RusticLuftig) [inline neurons/validator.py:656]: batches_to_use? len(batches_new) != len(batches_old)

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-bittensor
+bittensor==7.0.1
```
Comment (RusticLuftig) [inline requirements.txt:1]: In our testing for finetuning we struggled with anything > 6.9.3, so I'd recommend caution with this. Pinning is a good idea, but consider 6.9.3 (when we last checked, almost no other subnet used > 6.9.3). In particular, logging seems to be broken

File: neurons/validator.py (hunk 1 above)
Comment (alan-aboudib-mc) [reply neurons/validator.py:591]: I agree. In the original code, Release v3.1.4, there was no distinction between new and old losses. We introduced this distinction in 3.2.0 (the version with the bug), then I think that we should remove the distinction again. The way I understand it is that, if the model does not have metadata (line592) then we are no longer able to distinguish whether this is an old or new model, so we use whatever loss list (regardless, new or old), because it will be set to infinity.

If the model has meta data, the `losses` variable is completely overwritten in line 627.

File: neurons/validator.py (hunk 2 above)
Comment (alan-aboudib-mc) [reply neurons/validator.py:656]: Again, this was the case originally in v3.1.4, we introduced `batches_to_use` in our release instead, but now I am reverting it, for the exact same reason as my comment about `losses` above.

File: neurons/validator.py (hunk 1 above)
Comment (alan-aboudib-mc) [reply neurons/validator.py:591]: I agree, we discussed removed support to models prior to 7B, but for the next release

File: requirements.txt (hunk 3 above)
Comment (alan-aboudib-mc) [reply requirements.txt:1]: I need to look into that yes. Haven't tested with this version yet.

File: neurons/validator.py (hunk 1 above)
Comment (Sid-Data-Universe) [reply neurons/validator.py:591]: In the exceptional case line 627 does not get hit and you instead hit 641 which uses the losses as they are. Then in 655 when you compute wins you could have mismatched lengths of the losses.

File: pretrain/dataset.py [hunk 4]
```diff
@@ -257,8 +316,8 @@ def __init__(
         self,
         batch_size,
         sequence_length,
-        num_pages: typing.List[int],
-        tokenizer: AutoTokenizer,
+        num_pages=None,
+        tokenizer: AutoTokenizer=None,
```
Comment (RusticLuftig) [inline pretrain/dataset.py:320]: A tokenizer is required for this class to work (used on line 367), so None seems like an odd default

File: neurons/validator.py [hunk 5]
```diff
@@ -71,6 +71,8 @@ def state_path(self) -> str:
     def __init__(self):
         self.config = config.validator_config()
         bt.logging(config=self.config)
+        bt.logging.set_debug()
```
Comment (Sid-Data-Universe) [inline neurons/validator.py:74]: Is this still desired now that you have reverted to 6.9.3?

**Review by RusticLuftig — APPROVED** [review]
